File: model/src/main/java/com/itransition/fanfictionbackend/model/Comment.java
package com.itransition.fanfictionbackend.model;

import lombok.Data;

import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.Lob;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Data
@Entity
@Table(name = "comment")
public class Comment extends WithLongId {

    @Lob
    private String message;

    @ManyToOne(optional = false)
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne(optional = false)
    @JoinColumn(name = "fanfic_id")
    private Fanfic fanfic;
}

File: deprecated-app/src/main/kotlin/com/itransition/fanfictionbackend/model/Comment.kt
package com.itransition.fanfictionbackend.model

import javax.persistence.Entity
import javax.persistence.JoinColumn
import javax.persistence.Lob
import javax.persistence.ManyToOne
import javax.persistence.Table

@Entity
@Table(name = "comment")
class Comment : WithLongId() {

    @Lob
    var message: String? = null

    @ManyToOne(optional = false)
    @JoinColumn(name = "user_id")
    val user: User? = null

    @ManyToOne(optional = false)
    @JoinColumn(name = "fanfic_id")
    val fanfic: Fanfic? = null
}
